pkg/sources/github/repo.go

```diff
@@ -50,12 +50,15 @@ func (s *Source) cloneRepo(
 
 	case *sourcespb.GitHub_Token:
 		// We never refresh user provided tokens, so if we already have them, we never need to try and fetch them again.
+		s.userMu.Lock()
```

Comment:
If `s.userAndToken` panics, this will be locked forever, right? Do you think it's worth breaking out a new function (inline or otherwise) so you can use `defer`?